version: "3.9"
services:
  postgres:
    image: postgres:13.3
    environment:
      POSTGRES_HOST: postgres
      POSTGRES_DB: ${PGDATABASE}
      POSTGRES_USER: ${PGUSER}
      POSTGRES_PASSWORD: ${PGPASSWORD}
      POSTGRES_DATA: ${PGDATA}
    ports:
      - "${PGEXPOSEPORT}:5432"
    volumes:
      - ./dataDB:/var/lib/postgresql/data
    user: root
    healthcheck:
      test: [ "CMD", "/usr/bin/pg_isready", "-U", "${PGUSER}" ]
      interval: 10s
      timeout: 1s
      retries: 3
      start_period: 30s
    restart: always
  bot:
    image: agylam/bot
    environment:
      PGDATABASE: ${PGDATABASE}
      PGUSER: ${PGUSER}
      PGPASSWORD: ${PGPASSWORD}
      PGPORT: ${PGPORT}
      PGHOST: postgres
      TELEGRAM_TOKEN: ${TELEGRAM_TOKEN}
      DADATA_TOKEN: ${DADATA_TOKEN}
    depends_on:
      postgres:
        condition: service_healthy
    restart: always
    privileged: true

networks:
  default:
    driver: bridge
    driver_opts:
      com.docker.network.driver.mtu: 1400
